src/daemon/auth/providers/openid_connect/provider.rs

```diff
@@ -357,49 +363,84 @@ impl OpenIDConnectAuthProvider {
 
                                         match new_token_res {
                                             Ok(new_token) => {
-                                                return Some(Auth::Bearer(new_token));
+                                                // The new token was successfully acquired from the OpenID Connect Provider,
+                                                // and early returned.
+                                                return Ok(Auth::Bearer(new_token));
                                             }
                                             Err(err) => {
-                                                warn!(
-                                                    "OpenID Connect: Could not extend login session for user '{}': {}",
-                                                    &session.id, err
-                                                );
+                                                return Err(CoreErrorResponseType::Extension(format!(
+                                                    "Error while encoding the refreshed token {}",
```

Comment:
Suggestion: Prefix the error message with `krill_internal_error: ` to make it easy to distinguish from any actual provider error messages.